### PR TITLE
Test frame work pocs

### PR DESCRIPTION
**Pull request by keetron**
Containing adjusted Cucumber tests and Karate POC.

### CODE REVIEW

File: src/test/java/cucumber/GreetingServiceSteps.java [hunk 1]
```diff
@@ -39,6 +35,27 @@ public void theClientRequestsGETGreeting() throws Throwable{
...
+            throw r;
+        } catch (Exception e) {
+            System.out.println("Exception caught");
+            e.printStackTrace();
+        }
+
+        Assert.assertEquals("Did not receive ok response: ", HttpURLConnection.HTTP_OK, response.getStatus());
+    }
+
     public void theResponseShouldBeJSON(String jsonExpected){
```
Comment (mlvandijk) [inline src/test/java/cucumber/GreetingServiceSteps.java]: Refactor to remove code duplication.

File: src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java [hunk 2]
```diff
@@ -6,16 +6,17 @@
...
 
-    public GreetingServiceStepDefs(){
+    public GreetingServiceStepDefs() {
         Given("^the greeting service is running$", () -> {
-            SpringApplication.run(Application.class);
+            try {
+                SpringApplication.run(Application.class);
+            } catch (Throwable t) {
+                System.out.println("Throwable caught");
+            }
```
Comment (mlvandijk) [inline src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java]: Please add meaningful error messages

File: src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java [hunk 3]
```diff
@@ -27,6 +28,15 @@ public GreetingServiceStepDefs(){
             }
         });
 
+        When("^the client requests /GET /greeting using query string parameter User$", () -> {
+            try {
+                greetingServiceSteps.theClientRequestsGETGreetingUsingQueryStringParameterUser();
+            } catch (Throwable t) {
+                System.out.println("Throwable caught");
```
Comment (mlvandijk) [inline src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java]: Use capture group to pass parameter to method. Note: use (.*) to capture a String.

File: src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java (hunk 2 above)
Comment (keetron) [reply src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java]: Done, mostly as it is not always clear why a certain error message is in place.

File: src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java (hunk 3 above)
Comment (keetron) [reply src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java]: done, much easier than expected tbh

**Pull request merged**